paddle/phi/kernels/onednn/transpose_grad_kernel.cc

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "paddle/phi/kernels/transpose_grad_kernel.h"
+
+#include "paddle/phi/backends/onednn/onednn_reuse.h"
+#include "paddle/phi/core/kernel_registry.h"
+
+namespace phi {
+template <typename T, typename Context>
+void TransposeGradKernel(const Context& dev_ctx,
+                         const DenseTensor& out_grad,
+                         const std::vector<int>& axis,
+                         DenseTensor* x_grad) {
+  PADDLE_ENFORCE_EQ(dev_ctx.GetPlace().GetType() == phi::AllocationType::CPU,
+                    true,
+                    errors::PreconditionNotMet(
+                        "Operator DNNL TransposeGrad must use CPUPlace"));
+  if (!x_grad) return;
+
+  const auto& onednn_engine = dev_ctx.GetEngine();
+  std::vector<int> reversed_axis(axis);
+  int ndims = axis.size();
+  if (ndims == 1) {
+    Copy(dev_ctx, out_grad, out_grad.place(), false, x_grad);
```

Comment:
I found that `phi::Copy` and `framework::TensorCopy` behave inconsistently in the mkldnn scenario, you can still use `framework::TensorCopy` for the time being, the problem of `phi::Copy` I will solve